### ISSUE TITLE
Missing `name` parameter in chat completion message parameters.

### ISSUE THREAD

**Issue by anzemur** [openai/openai-openapi#118]
According to [the documentation](https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages) we should be able to specify the name for each of the base message types/roles (system, assistant, and user) to provide the model with additional information to differentiate between participants of the same role (useful for few-shot prompting).

**Comment by rattrayalex** [comment]
cc @nknj – thoughts on adding this?

**Comment by anzemur** [comment]
@rattrayalex @nknj  any more information on this? I came upon this issue when migrating from version 3.x.x and it seems like this parameters are not deprecated.

**Comment by transitive-bullshit** [comment]
It looks like they're deprecating `name` support, but it would be nice to get an official explanation.

**Comment by logankilpatrick** [comment]
Can you confirm what the issue is? We document these in the API, are you saying they are not supported here in the SDK?

**Comment by anzemur** [comment]
Yes, they are not supported in the SDK - types `ChatCompletionSystemMessageParam`,  `ChatCompletionUserMessageParam`, and `ChatCompletionAssistantMessageParam` should all include the `name` parameter but the usage of these types would result in the following error: `Object literal may only specify known properties, and ''name'' does not exist in type`.

I am using the node library v4.18.0.

**Comment by logankilpatrick** [comment]
This should be fixed soon! Hang tight.

**Comment by davidmigloz** [comment]
https://github.com/openai/openai-openapi/pull/126 includes the fix, so, when merged, https://github.com/openai/openai-openapi/pull/119 can be closed

**Comment by sp88011** [comment]
Same (?) issue here. I'm using `4.20.1` and am getting this error when calling `openai.chat.completions.create(...)`

```
  Types of property 'messages' are incompatible.
        Type '{ role: ChatCompletionRole; content: string; }[]' is not assignable to type 'ChatCompletionMessageParam[]'.
          Type '{ role: OpenAI.Chat.Completions.ChatCompletionRole; content: string; }' is not assignable to type 'ChatCompletionMessageParam'.
            Property 'name' is missing in type '{ role: OpenAI.Chat.Completions.ChatCompletionRole; content: string; }' but required in type 'ChatCompletionFunctionMessageParam'.
            ```

**Comment by rattrayalex** [comment]
Sorry about this! A true fix has merged on this repo and should go out on the Node SDK shortly.

**Comment by shamoons** [comment]
Still seems to be broken?

**Comment by rattrayalex** [comment]
@shamoons where do you see `name` missing that you would have expected it?
